src/engine/source/schemf/src/schema.cpp

```diff
@@ -118,16 +137,48 @@ const Field& Schema::get(const DotPath& name) const
 bool Schema::hasField(const DotPath& name) const
 {
     const auto* current = &m_fields;
+    auto isParentSchema = false;
     for (auto it = name.cbegin(); it != name.cend(); ++it)
     {
         auto entry = current->find(*it);
         if (entry == current->end())
         {
-            return false;
+            if (!isParentSchema)
+            {
+                return false;
+            }
+
+            throw std::runtime_error(fmt::format("Field '{}' does not exist in '{}'", it->data(), name.str()));
         }
 
+        isParentSchema = true;
```

Comment:
question: Why do you set this variable and then not use it?

Reply:
It is used